src/models/polar.jl

```diff
@@ -0,0 +1,509 @@
+                                  polar.ybus_im,
+                                  pbus, qbus, pv, pq, nbus)
+    return F
+end
+
+# TODO: find better naming
+function init_ad_factory(polar::PolarForm{T, VT, AT}, x, u, p) where {T, VT, AT}
+    nbus = PS.get(polar.network, PS.NumberOfBuses())
+    ngen = PS.get(polar.network, PS.NumberOfGenerators())
+    npv = PS.get(polar.network, PS.NumberOfPVBuses())
+    npq = PS.get(polar.network, PS.NumberOfPQBuses())
+    n_states = get(polar, NumberOfState())
+    # Indexing
+    ref = convert(polar.AT{Int, 1}, polar.network.ref)
+    pv = convert(polar.AT{Int, 1}, polar.network.pv)
+    pq = convert(polar.AT{Int, 1}, polar.network.pq)
+    # Network state
+    Vm, Va, pbus, qbus = get_network_state(polar, x, u, p)
+    F = VT(undef, n_states)
+    fill!(F, zero(T))
+    # Build the AD Jacobian structure
+    stateJacobianAD = AD.StateJacobianAD(F, Vm, Va,
+                                         polar.ybus_re, polar.ybus_im, pbus, qbus, pv, pq, ref, nbus)
+    designJacobianAD = AD.DesignJacobianAD(F, Vm, Va,
+                                           polar.ybus_re, polar.ybus_im, pbus, qbus, pv, pq, ref, nbus)
+    return stateJacobianAD, designJacobianAD
+end
+
+function jacobian(polar::PolarForm, jac::AD.AbstractJacobianAD, x, u, p)
+    nbus = PS.get(polar.network, PS.NumberOfBuses())
+    ngen = PS.get(polar.network, PS.NumberOfGenerators())
+    # Indexing
+    ref = convert(polar.AT{Int, 1}, polar.network.ref)
+    pv = convert(polar.AT{Int, 1}, polar.network.pv)
+    pq = convert(polar.AT{Int, 1}, polar.network.pq)
+    # Network state
+    Vm, Va, pbus, qbus = get_network_state(polar, x, u, p)
+    AD.residualJacobianAD!(jac, residualFunction_polar!, Vm, Va,
+                           polar.ybus_re, polar.ybus_im, pbus, qbus, pv, pq, ref, nbus, TIMER)
+    return jac.J
+end
+
+function powerflow(
+    polar::PolarForm{T, VT, AT},
+    jacobian::AD.StateJacobianAD,
+    x::VT,
+    u::VT,
+    p::VT;
+    npartitions=2,
+    solver="default",
+    preconditioner=Precondition.NoPreconditioner(),
+    tol=1e-7,
+    maxiter=20,
+    verbose_level=0,
+) where {T, VT, AT}
+    # Retrieve parameter and initial voltage guess
+    @timeit TIMER "Init" begin
+        Vm, Va, pbus, qbus = get_network_state(polar, x, u, p)
+    end
+    V = convert(polar.AT{Complex{T}, 1}, polar.network.vbus)
+
+    nbus = PS.get(polar.network, PS.NumberOfBuses())
+    ngen = PS.get(polar.network, PS.NumberOfGenerators())
+    npv = PS.get(polar.network, PS.NumberOfPVBuses())
+    npq = PS.get(polar.network, PS.NumberOfPQBuses())
+    n_states = get(polar, NumberOfState())
+
+    # TODO: avoid moving pv and pq each time we are calling powerflow
+    ref = convert(polar.AT{Int, 1}, polar.network.ref)
+    pv = convert(polar.AT{Int, 1}, polar.network.pv)
+    pq = convert(polar.AT{Int, 1}, polar.network.pq)
+
+    # iteration variables
+    iter = 0
+    converged = false
+
+    # indices
+    j1 = 1
```

Comment:
At some point, the indexing SHOULD be specified in the formulation `PolarForm` directly. But I am still figuring out a clean way to do so